editoast/src/schema/rolling_stock_schema/rolling_stock.rs

```diff
@@ -68,145 +69,51 @@ pub struct RollingStockMetadata {
-    #[serde(flatten)]
-    pub rolling_stock: LightRollingStock,
-    pub liveries: Vec<RollingStockLiveryMetadata>,
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct EffortCurve {
+    speeds: Vec<f64>,
+    max_efforts: Vec<f64>,
 }
 
-impl LightRollingStock {
-    /// List the rolling stocks without their effort curves
-    pub async fn list(
-        db_pool: Data<DbPool>,
-        page: i64,
-        per_page: i64,
-    ) -> Result<PaginatedResponse<LightRollingStock>> {
-        sql_query(
-            "WITH liveries_by_rs AS (SELECT rolling_stock_id, json_build_object('id', livery.id, 'name', livery.name) AS liveries
-            FROM osrd_infra_rollingstocklivery livery)
-            SELECT
-                rolling_stock.*,
-                COALESCE(ARRAY_AGG(liveries_by_rs.liveries), ARRAY[]::json[]) AS liveries
-            FROM osrd_infra_rollingstock rolling_stock
-            LEFT JOIN liveries_by_rs liveries_by_rs ON liveries_by_rs.rolling_stock_id = rolling_stock.id
-            GROUP BY rolling_stock.id"
-        )
-        .paginate(page)
-        .per_page(per_page)
-        .load_and_count(db_pool)
-        .await
-    }
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct ModeEffortCurves {
+    curves: Vec<ConditionalEffortCurve>,
+    default_curve: EffortCurve,
+    is_electric: bool,
+}
 
-    /// Retrieve a rolling stock without its effort curves
-    pub async fn retrieve(
-        db_pool: Data<DbPool>,
-        rolling_stock_id: i64,
-    ) -> Result<LightRollingStockWithLiveries> {
-        block::<_, Result<_>>(move || {
-            use crate::tables::osrd_infra_rollingstock::dsl as rolling_stock_dsl;
-            use crate::tables::osrd_infra_rollingstocklivery::dsl as livery_dsl;
-            let mut conn = db_pool.get()?;
-            let rolling_stock = match rolling_stock_dsl::osrd_infra_rollingstock
-                .find(rolling_stock_id)
-                .select(LightRollingStock::as_select())
-                .first(&mut conn)
-            {
-                Ok(rolling_stock) => Ok::<_, InternalError>(rolling_stock),
-                Err(DieselError::NotFound) => {
-                    Err(RollingStockError::NotFound { rolling_stock_id }.into())
-                }
-                Err(e) => Err(e.into()),
-            }?;
-            let liveries = livery_dsl::osrd_infra_rollingstocklivery
-                .filter(livery_dsl::rolling_stock_id.eq(rolling_stock_id))
-                .select(RollingStockLiveryMetadata::as_select())
-                .load(&mut conn)?;
-            Ok(LightRollingStockWithLiveries {
-                rolling_stock,
-                liveries,
-            })
-        })
-        .await
-        .unwrap()
-    }
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct EffortCurves {
+    modes: HashMap<String, ModeEffortCurves>,
+    default_mode: String,
 }
 
+// Errors
+
```

Comment:
Errors could be moved to the view since we should need them in the schema module.
The idea is to keep as possible things private.